src/components/Gauge.js

```diff
@@ -76,119 +53,146 @@ const StatusGaugeWrapper = styled.div`
   }
 `
 
-const GaugeTotalText = styled.text`
-  font-family: ${font};
-  font-size: ${props => px(props.count > 999 ? 20 : 26)};
-  fill: #5e5e5e;
-  letter-spacing: -1px;
-  text-anchor: middle;
-  transform: translateX(0);
-`
+class StatusGauge extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.DoughnutChart = null
+  }
+
+  componentDidMount () {
+    this.generateDoughnut(this.props)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.generateDoughnut(nextProps)
+  }
+
+  componentWillUnmount () {
+    if (this.DoughnutChart) {
+      this.DoughnutChart.destroy()
```

Comment:
We should also assign it to `null` after calling `destroy()`, otherwise would be called for a second time in `generateDoughnut()` if the components mounts again.
We could actually have a `destroy()` method as part of the component, encapsulating the whole `if`.